src/journal/journald-kmsg.c

```diff
@@ -384,6 +384,17 @@ int server_open_dev_kmsg(Server *s) {
         if (!s->read_kmsg)
                 return 0;
 
+        if (!s->boot_kmsg) {
+                /* clear out /dev/kmsg, we don't want all its messages */
+                char buffer[40960];
+                while (1) {
+                        int ret;
+                        ret = read(s->dev_kmsg_fd, buffer, 40960);
+                        if (ret <= 0)
+                                break;
+                }
+        }
```

Comment:
we have flush_fd() for this kind of stuff, which does the same

Reply:
this doesn't look right in general, i.e. when journald is restarted we would run this and drop everything happening during the restart. 

The kernel has a seqnum concept which allows us to identify which records we already have seen and which ones not. if the feature you are looking for is indeed desirable it should be built on that: see journald-kmsg.c, in dev_kmsg_record() and  server_open_kernel_seqnum(), where we compare and manage seqnums. We store the last msg seqnum we saw in a file /run/kernel-seqnum. Most likely it suffices to initialize that one after the first call to server_flush_dev_kmsg() and leave it unset before that.

Reply:
Thanks, I'll look into that